src/cobalt.rs

```diff
@@ -133,15 +133,40 @@ fn generate_doc(
 ) -> Result<()> {
     // Everything done with `globals` is terrible for performance.  liquid#95 allows us to
     // improve this.
-    let mut globals: liquid::value::Object = vec![
-        (
-            "site".into(),
-            liquid::value::Value::Object(context.site.clone()),
-        ),
-        global_collection,
-    ]
-    .into_iter()
-    .collect();
+    let mut globals: liquid::value::Object =
+        if doc.front.pagination.is_some() || doc.front.pagination_compat {
+            warn!(
+            "`collections.posts` is deprecated. Please transition to `paginator` and then disable \
+            support for `collections.posts` by setting `pagination_compat` to `false`"
+            );
+            let collections_posts: liquid::value::Object =
+                vec![("posts".into(), global_collection.1.clone())]
+                    .into_iter()
+                    .collect();
+            vec![
+                (
+                    "site".into(),
+                    liquid::value::Value::Object(context.site.clone()),
+                ),
+                (
+                    "collections".into(),
+                    liquid::value::Value::Object(collections_posts),
+                ),
+                global_collection,
+            ]
+            .into_iter()
+            .collect()
+        } else {
+            vec![
+                (
+                    "site".into(),
+                    liquid::value::Value::Object(context.site.clone()),
+                ),
+                global_collection,
+            ]
+            .into_iter()
+            .collect()
+        };
```

Comment:
(picked arbitrary place because GH limitations)

Shouldn't this change be leading us to no longer need `generate_collections_var`?

Reply:
If the pagination is not activated, we still need it, isn't it?

Reply:
I understand now (https://github.com/cobalt-org/cobalt.rs/issues/681#issuecomment-528568537)

Reply:
`generate_collections_var` is used in `generate_pots` and I still use it to manage the case where no paginator was created (because of no posts, test `data_files` was failing)